packages-common/pluggable-widgets-generator/generators/app/index.js

```diff
@@ -0,0 +1,284 @@
+const { join } = require("path");
+const pkg = require(join(__dirname, "/../../package.json"));
+const { mkdir } = require("fs").promises;
+const Generator = require("yeoman-generator");
+
+const promptTexts = require("./lib/prompttexts.js");
+const text = require("./lib/text.js");
+const utils = require("./lib/utils.js");
+
+const widgetSrcFolder = "src/components/";
+
+const banner = text.getBanner(pkg);
+
+class MxGenerator extends Generator {
+    constructor(args, opts) {
+        super(args, opts);
+
+        /**
+         * Test if a widget folder was passed by argument and try to create the new folder
+         */
+        if (args.length > 0) {
+            const dir = args.map(arg => arg.replace(/(^|\s)\S/g, l => l.toLowerCase())).join("-");
+            const name = args.map(arg => arg.replace(/(^|\s)\S/g, l => l.toUpperCase())).join("");
+            if (dir) {
+                this.dir = dir;
+                this.widgetParamName = name;
+            }
+        }
+    }
+
+    async initializing() {
+        this.FINISHED = false;
+
+        if (!this.dir && !(await utils.isDirEmpty(this.destinationRoot()))) {
+            this.log(banner);
+            this.log(text.DIR_NOT_EMPTY_ERROR);
+            this.FINISHED = true;
+        } else if (this.dir && utils.dirExists(this.dir) && !(await utils.isDirEmpty(this.dir))) {
+            this.log(banner);
+            this.log(text.DIR_NOT_EMPTY_ERROR);
+            this.FINISHED = true;
+        }
+    }
+
+    async prompting() {
+        if (this.FINISHED) {
+            return;
+        }
+
+        this.log(banner);
+
+        const dir = await utils.findMprDir(this.dir);
+
+        const widgetAnswers = await this.prompt(promptTexts.promptWidgetProperties(dir, this.widgetParamName));
+        const testAnswers = await this.prompt(promptTexts.promptTestsInfo(widgetAnswers));
+        const combinedAnswers = Object.assign(widgetAnswers, testAnswers);
+
+        this.widget = utils.getWidgetDetails(combinedAnswers);
+    }
+
+    async writing() {
+        if (this.dir) {
+            if (!utils.dirExists(this.dir)) {
+                await mkdir(this.dir);
+            }
+            this.destinationRoot(this.dir);
+        }
+        this._writePackage();
+        this._writeWidgetXML();
+        this._writeWidgetFiles();
+        this._writeCompilerOptions();
+        this._writeUtilityFiles();
+        this._writeUnitTests();
+        this._writeEndToEndTests();
+    }
+
+    install() {
+        if (this.FINISHED) {
+            return;
+        }
+        this.log(text.INSTALL_FINISH_MSG);
+        this.npmInstall();
+    }
+
+    end() {
+        if (this.FINISHED) {
+            return;
+        }
+
+        if (utils.isDirEmpty(this.destinationPath("node_modules"))) {
+            this.log(text.END_NPM_NEED_INSTALL_MSG);
+        } else {
+            this.log(text.END_RUN_BUILD_MSG);
+            this.spawnCommandSync("npm", ["run", "lint"]); // eslint-disable-line no-sync
```

Comment:
can't you make these async and get rid of eslint disable?

Reply:
Yeoman doesn't have any builtin support for this, so we would have to reuse the test code for PIW-tools if you want that (`execAsync`). This will require two extra dependencies though: `util` and `child_process`. Do you still want to change it?